smart-contracts/contracts/utils/LockProxy.sol

```diff
@@ -0,0 +1,954 @@
+     * https://eth.wiki/json-rpc/API#eth_getstorageat[`eth_getStorageAt`] RPC call.
+     * `0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103`
+     */
+    function _dispatchAdmin() private returns (bytes memory) {
+        _requireZeroValue();
+
+        address admin = _getAdmin();
+        return abi.encode(admin);
+    }
+
+    /**
+     * @dev Returns the current implementation.
+     *
+     * TIP: To get this value clients can read directly from the storage slot shown below (specified by EIP1967) using the
+     * https://eth.wiki/json-rpc/API#eth_getstorageat[`eth_getStorageAt`] RPC call.
+     * `0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`
+     */
+    function _dispatchImplementation() private returns (bytes memory) {
+        _requireZeroValue();
+
+        address implementation = _implementation();
+        return abi.encode(implementation);
+    }
+
+    /**
+     * @dev Changes the admin of the proxy.
+     *
+     * Emits an {AdminChanged} event.
+     */
+    function _dispatchChangeAdmin() private returns (bytes memory) {
+        _requireZeroValue();
+
+        address newAdmin = abi.decode(msg.data[4:], (address));
+        _changeAdmin(newAdmin);
+
+        return "";
+    }
+
+    /**
+     * @dev Upgrade the implementation of the proxy.
+     */
+    function _dispatchUpgradeTo() private returns (bytes memory) {
+        _requireZeroValue();
+
+        address newImplementation = abi.decode(msg.data[4:], (address));
+        _upgradeToAndCall(newImplementation, bytes(""), false);
+
+        return "";
+    }
+
+    /**
+     * @dev Upgrade the implementation of the proxy, and then call a function from the new implementation as specified
+     * by `data`, which should be an encoded function call. This is useful to initialize new storage variables in the
+     * proxied contract.
+     */
+    function _dispatchUpgradeToAndCall() private returns (bytes memory) {
+        (address newImplementation, bytes memory data) = abi.decode(msg.data[4:], (address, bytes));
+        _upgradeToAndCall(newImplementation, data, true);
+
+        return "";
+    }
+
+    /**
+     * @dev Returns the current admin.
+     *
+     * CAUTION: This function is deprecated. Use {ERC1967Upgrade-_getAdmin} instead.
+     */
+    function _admin() internal view virtual returns (address) {
+        return _getAdmin();
+    }
+
+    /**
+     * @dev To keep this contract fully transparent, all `ifAdmin` functions must be payable. This helper is here to
+     * emulate some proxy functions being non-payable while still allowing value to pass through.
+     */
+    function _requireZeroValue() private {
+        require(msg.value == 0);
+    }
+}
+
+
+// File contracts/utils/LockProxy.sol
+
+interface ILockProxy is ITransparentUpgradeableProxy {}
+
+contract LockProxy is TransparentUpgradeableProxy {
```

Comment:
yeah it looks like we could just use `ITransparentUpgradeableProxy` directly?

Reply:
yes we can remove the concept of `LockProxy` and keep only the native (flatten) `TransparentUpgradeableProxy` from Open Zeppelin